pkg/occlient/occlient_test.go

```diff
@@ -8,9 +8,72 @@ import (
 	"testing"
 
 	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
 	ktesting "k8s.io/client-go/testing"
 )
 
+func TestCreatePVC(t *testing.T) {
+	tests := []struct {
+		name    string
+		size    string
+		labels  map[string]string
+		wantErr bool
+	}{
+		{
+			name: "storage 10Gi",
+			size: "10Gi",
+			labels: map[string]string{
+				"name":      "mongodb",
+				"namespace": "blog",
+			},
+			wantErr: false,
+		},
+		{
+			name: "storage 1024",
+			size: "1024",
+			labels: map[string]string{
+				"name":      "PostgreSQL",
+				"namespace": "backend",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+
+			_, err := fkclient.CreatePVC(tt.name, tt.size, tt.labels)
+
+			// Checks for error in positive cases
+			if !tt.wantErr == (err != nil) {
+				t.Errorf(" client.CreatePVC(name, size, labels) unexpected error %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			// Check for validating actions performed
+			if len(fkclientset.Kubernetes.Actions()) != 1 {
+				t.Errorf("expected 1 action in CreatePVC got: %v", fkclientset.RouteClientset.Actions())
+			}
+			// Checks for return values in positive cases
+			if err == nil {
+				createdPVC := fkclientset.Kubernetes.Actions()[0].(ktesting.CreateAction).GetObject().(*corev1.PersistentVolumeClaim)
+				// fmt.Println(createdPVC)
+				// created route should be labeled with labels passed to CreatePVC
+				if !reflect.DeepEqual(createdPVC.Labels, tt.labels) {
+					t.Errorf("labels in created route is not matching expected labels, expected: %v, got: %v", tt.labels, createdPVC.Labels)
+				}
+				//  name and size that route is pointg to should match
+				//  if createdPVC.Size != tt.size {
+				//      t.Errorf("route is not matching to expected service name, expected: %s, got %s", tt.size, createdPVC.Size)
+				//  }
```

Comment:
remove these lines of code?

Reply:
sorry this is a WIP and i still have to add checks for the size of createdPVC